File: advent-of-code/2021/day01/day01b.py
#!/usr/bin/env python

"""Advent of Code 2021, day 1, part B"""

import fileinput
import itertools

import more_itertools


triples = more_itertools.windowed(map(int, fileinput.input()), 3)
pairs = itertools.pairwise(map(sum, triples))
print(sum(pre < cur for pre, cur in pairs))

File: advent-of-code/2022/day01/day01b.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Advent of Code, day 1, part B

$VERBOSE = true

def run
  puts ARGF.read
           .split(/\n{2,}/)
           .map(&:strip)
           .reject(&:empty?)
           .map { |group| group.split.map(&:to_i).sum }
           .sort
           .last(3)
           .sum
end

run if $PROGRAM_NAME == __FILE__
